cincinnati/src/plugins/mod.rs

```diff
@@ -364,6 +364,67 @@ where
     io.try_into()
 }
 
+/// Wrapper around `process` with an optional timeout.
+///
+/// It creates a new runtime per call which is moved to a new thread.
+/// This has the desired effect of timing out even if the runtime is blocked by a task.
+/// It has the sideeffect that these threads are unrecoverably leaked.
+///
+/// These two strategies are used in combination to implement the timeout:
+/// 1. Use the runtime's internal timeout implementation which works for proper async tasks.
+/// 2. Spawn a separate sleeper thread in case the first timeout doesn't work.
+pub fn process_blocking<T>(
+    plugins: T,
+    initial_io: PluginIO,
+    timeout: Option<std::time::Duration>,
+) -> Fallible<InternalIO>
+where
+    T: Iterator<Item = &'static BoxedPlugin>,
+    T: Sync + Send,
+    T: 'static,
+{
+    let mut runtime = tokio::runtime::Runtime::new()?;
+
+    let timeout = match timeout {
+        None => return runtime.block_on(process(plugins, initial_io)),
+        Some(timeout) => timeout,
+    };
+
+    let (tx, rx) = std::sync::mpsc::channel::<Fallible<InternalIO>>();
+
+    {
+        let tx = tx.clone();
+
+        std::thread::spawn(move || {
```

Comment:
IIUC this is necessary so that `release_scraper_dockerv2` could do `tokio::task::spawn_blocking(move || assemble_metadata(&blob, metadata_filename))
            .await`?
Is it necessary to spawn a blocking thread there?

Reply:
The code here tries to deal with the worst case which would be a plugin which completely blocks the processing. The split into two threads here is to have a reliable timeout source.